src/Test/AbstractIntegrationTestCase.php

```diff
@@ -269,14 +262,27 @@ function (FixerInterface $fixer) {
                 new NullCacheManager()
             );
 
+            Tokens::clearCache();
             $runner->fix();
             $fixedInputCodeWithReversedFixers = file_get_contents($tmpFile);
 
-            $this->assertNotSame(
-                $fixedInputCode,
-                $fixedInputCodeWithReversedFixers,
-                sprintf('Set priorities must be significant. If fixers used in reverse order return same output then the integration test is not sufficient or the priority relation between used fixers should not be set. In "%s".', $case->getFileName())
-            );
+            // If output is different depends on rules order - we need to verify that the rules are ordered by priority.
+            // If not, any order is valid.
+            if ($fixedInputCode !== $fixedInputCodeWithReversedFixers) {
+                $this->assertGreaterThan(
+                    1,
+                    count(array_unique(array_map(
+                        function (FixerInterface $fixer) {
+                            return $fixer->getPriority();
+                        },
+                        $fixers
+                    ))),
+                    sprintf(
+                        'Rules priorities are not differential enough. If rules would be used in reverse order then final output would be different than the expected one. For that, different priorities must be set up for used rules to ensure stable order of them. In "%s".',
+                        $case->getFileName()
+                    )
+                );
+            }
```

Comment:
no. if we have 2 fixers with different priorities, itest will perform fixing process using different order